### PR TITLE
Top 5 merchants by revenue

### PR DESCRIPTION
**Pull request by MadelineMauser**
- Adds a class method to the merchant model to retrieve the top 5 merchants by total revenue
- Adds an instance method to the merchant model to calculate total revenue to help with testing
- Adds a section to the admin merchants index that displays the top 5 merchants along with their total revenue and a link to each merchant's show page
- One unrelated test is not passing, but Will caught the mistake and made changes that should fix it once his PR is merged

### CODE REVIEW

**Review by kenzjoy — APPROVED** [review]
woooo awesome we are so close!

File: app/models/merchant.rb [hunk 1]
```diff
@@ -37,4 +37,18 @@ def top_5_items
         .limit(5)
         .sum('quantity * invoice_items.unit_price')
   end
+
+  def self.top_5_by_revenue
+    joins(:transactions)
+    .where(transactions: {result: 1})
```
Comment (msakr21) [inline app/models/merchant.rb:43]: I think we need to add a condition for which invoices to consider for the Revenue (we wouldn't want in progress or cancelled invoices.)

**Review by msakr21 — COMMENTED** [review]
If everyone can check my comment and weigh in with their thoughts that would be much appreciated!

File: app/models/merchant.rb (hunk 1 above)
Comment (kenzjoy) [reply app/models/merchant.rb:43]: agreed good eye Mostafa, our query king!

Comment (willjw2) [reply app/models/merchant.rb:43]: Do you think we'd need that even though we're already filtering for successful transactions? I'm not super clear on what the relationship between transaction status and invoice status would be

**Review by willjw2 — APPROVED** [review]
Looks good, great job on adding the additional where clause

**Pull request merged**